### DISCUSSION TITLE
Settings of Sodium fire in FDS

### DISCUSSION THREAD

**Discussion by lihy213** [firemodels/fds#13054] [Q&A]
Dear developers,

I want to set up a combustion reaction of liquid metal sodium in FDS, where the temperature of the sodium is about 500°C and the pressure is 2 atm., and the reaction is the combustion of the sodium in contact with oxygen. But setting up the fuel as sodium directly in REAC line prompts a simple chemical reaction containing only C, H, O, etc. I tried setting up the reactants, oxidizers, and products separately in the SPEC line in lumped species. I'm not sure if this is the right way to do it. Can you please answer how I should set this up correctly?
My case is setted as follow,
[sodium_fire_pan.txt](https://github.com/user-attachments/files/15804380/sodium_fire_pan.txt)

Regards

LI

**Comment by drjfloyd** [comment]
User's Guide 13.2

**Reply by lihy213** [reply]
I met a problem, that is, in the experiment, 550 kg of liquid sodium was injected onto a $2.25 m^2$ flat plate for combustion in 210 s.  In applying a pyrolysis model to liquid sodium, the density of the liquid sodium is specified in MATL; is the thickness specified in the SURF line derived from `mass/density/area`($\frac{m}{\rho A}$)?
**The latest**
```fortran
&MATL ID='SODIUM LIQUID',
      SPECIFIC_HEAT=1.3,
      CONDUCTIVITY=148.0,
      DENSITY=968.0,
      HEAT_OF_REACTION=4673.91,
      BOILING_TEMPERATURE=882.9,
      EMISSIVITY=0.03575,
      SPEC_ID(1,1)='SODIUM',
      NU_SPEC(1,1)=1.0/ 
&SURF ID='SODIUM POOL',
      RGB=231,28,13,
      TMP_INNER=505.0,
      MATL_ID(1,1)='SODIUM LIQUID',
      MATL_ID(2,1)='STEEL',
      MATL_MASS_FRACTION(1,1)=1.0,
      THICKNESS= 0.25,0.003,
      SPEC_ID='SODIUM',
      BACKING = 'INSULATED'/
```
**Previous**
```fortran
&SURF ID='SODIUM POOL',
      RGB=231,28,13,
      TMP_INNER=505.0,
      MATL_ID(1,1)='SODIUM LIQUID',
      MATL_ID(2,1)='STEEL',
      MATL_MASS_FRACTION(1,1)=1.0,
      THICKNESS= 0.25,0.003,
      SPEC_ID='SODIUM',
      BACKING = 'INSULATED'/
&SURF ID='SODIUM POOL',
      RGB=231,28,13,
      HEAT_TRANSFER_COEFFICIENT=0.0,
      TMP_INNER=505.0,
      MATL_ID(1,1)='SODIUM LIQUID',
      MATL_MASS_FRACTION(1,1)=1.0,
      MASS_FLUX=1.16,
      RAMP_MF = 'SODIUM INJECT',
      THICKNESS(1)= 0.05,
      SPEC_ID='SODIUM'/
&RAMP ID='SODIUM INJECT'	 T=       	0	 F=       	1	 /
&RAMP ID='SODIUM INJECT'	 T=       	10	 F=       	1	 /
&RAMP ID='SODIUM INJECT'	 T=       	210	 F=       	1	 /
&RAMP ID='SODIUM INJECT'	 T=       	210.1	 F=       	0	 /
&RAMP ID='SODIUM INJECT'	 T=       	1000	 F=       	0	 /
```
I was previously given MASS_FLUX in the SURF line, but this is the mass flux of the gas and should not be used as a substitute for the mass injection rate of sodium in the experiment?
But after removing the MASS_FLUX, based on the calculated thickness, the pyrolysis reaction is performed and the HRR is almost very small, about 0.04 kW. How should I proceed to meet the mass requirement of the experiment and at the same time make the liquid sodium burn up?
![图片](https://github.com/user-attachments/assets/691067ac-c2e7-42e5-9931-a337f0bb59c6)

**Reply by drjfloyd** [reply]
A liquid pool sodium fire releases heat both by surface oxidation of the liquid sodium and evaporation of sodium vapor that then burns in the vapor phase. You only have one of these. You may want to reseach sodium fire chemistry. There are existing computer codes and methods developed as part of liquid sodium nuclear reactors in the US. Those manuals may give you some insights (for example SOFIRE).

You are correct that MASS_FLUX and RAMP_MF are for the gas. FDS does not have a second fluid phase model for modeling the flow and internal circulation of pools. The way I see it there are three possible conditions:

1. The sodium burning rate is sigificantly less than the addition rate, i.e., when the sodium addition ends only a few percent has burned. In this case may be sufficient to just start with the full pool depth.
2.  The sodium burning rate is similar to the addition rate, i.e., you do not form a deep pool of sodium. In this case you might be able to use lagrangian particles with SURF.
3. The sodium burning rate is between 1 and 2.  Here you will form a pool of some depth but the addition time is a significant fraction of the burning time. If you want to predict this condition, it may be that you need a different tool than FDS. Offhand I am not sure of a good approach for this.

Each of these are not typical applications of FDS. We cannot tell you what inputs you would need for any of those options. This will require some research on your part to develop a method appropirate for your specific needs.

**Reply by lihy213** [reply]
Thanks. So the input lines I provided just represent the process of evaporation of liquid sodium but no trigger to burn it up, right? Could it be appropriate to add an ignition temperature to burn it up? And it seems that it's reasonable for your suggestions 1 and 2, but the question is the way to implement it in FDS input file. Sadly, I have do some researches for liquid sodium fire, and indeed some codes could solve this situation, but my boss ask me to simulate the process by FDS because of its great performance in fire and three dimensional calculation(others in nuclear are zero or one dimension). Maybe the secondary development is needed in FDS to realize it. :(

**Reply by lihy213** [reply]
Now I am stuck in this situation.

**Reply by drjfloyd** [reply]
Your input lines do not contain all the mechanism via which sodium oxidizes. You have no MATL reaction for direct oxidation on the liquid surface. I cannot tell you what those inputs should be, I do not know the kinetics of sodium oxidation. That is something you will need to research and determine.

**Reply by lihy213** [reply]
Now the question is that the liquid sodium is evaporated into gas but not burn up. If I set the specified MASS_FLUX, it fires, despite it doesn't satisfy the real process.

**Reply by lihy213** [reply]
Another question: I specify the TMP_FRONT or TMP_INNER in the SURF line and monitor the WALL TEMPERATURE through a DEVC with a SURF_ID of the specified SURF, the wall temperature begins with TMPA rather the TMP_FRONT or TMP_INNER.

**Reply by drjfloyd** [reply]
This is what the User's Guide says those inputs do. Deciding which one is appropriate for your specific needs is a decision for you to make based on how you ultimately decide to model the sodium fire.

8.2.1 Specified Solid Surface Temperature

the input TMP_FRONT, the surface temperature in units of ◦C, can be used to specify a fixed
temperature boundary condition

8.3.4 Initial Solid Temperature

By default, the initial temperature of the solid material is set to ambient (TMPA on the MISC line). Use
TMP_INNER on the SURF line to specify a different initial temperature of the solid.
